BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -20,13 +20,44 @@ public BlogDataStore(IFileSystem fileSystem)
         {
             _fileSystem = fileSystem;
             InitStorageFolder();
+            InitCurrentId();
         }
         public void InitStorageFolder()
         {
             _fileSystem.CreateDirectory(StorageFolder);
             _fileSystem.CreateDirectory(DraftsFolder);
         }
 
+        public void InitCurrentId()
+        {
+            int max = 0;

```

Comment:
you should have a guard in this method to make sure it doesn't get called separately by multiple threads.  Something probably like this:

    if(CurrentId == 0)  // only do this once
    {
        lock(lockObj)
        {
            if(CurrentId == 0) // make sure you're still the first person inside the lock
            { /* everything else in this method goes here */ }